### PR TITLE
Added individual charity event page

### PR DESCRIPTION
**Pull request by cezaris13**
Added individual page, where user can see their event:
image
By clicking "edit event" button user is brought to edit page, where item can be updated and deleted: 
image
Updated main page:
image
Made those items clickable. Finally, changed Id type from `int` to `Guid` since it is easier to create a new , unique `guid` than `int` value.

Update:
Added 3 dots settings which redirects us straight to the edit menu:
image
Added alert message which shows what has been done:
image
Also, removed unneeded files from original blazor project(fetchData.razor, about page, SurveyPrompt.razor)

### CODE REVIEW

File: Pages/EditCharityEvent.razor [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿@page "/editcharityevent/{Id:guid}"
+@using Karma.Models
+@using Karma.Data
+@inject NavigationManager UriHelper
```
Comment (cezaris13) [inline Pages/EditCharityEvent.razor:4]: inconsistent naming.

File: Pages/AddCharityEvent.razor [hunk 2]
```diff
@@ -2,6 +2,7 @@
 @using Karma.Models
 @using Karma.Data
 @inject IDataModificator dataModificator
```
Comment (cezaris13) [inline Pages/AddCharityEvent.razor:4]: inconsistent naming

File: Pages/ShowIndividualEvent.razor [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <h2>@charityEvent.Description</h2>
+
+    <h3>@charityEvent.State.ToString()</h3>
+
+    <button class="btn btn-primary" @onclick="MoveToIndividualEvent">Update event</button>
+}
+else
+{
+    <h1>Sorry,but this event was not found</h1>
+}
```
Comment (Mart1nux) [inline Pages/ShowIndividualEvent.razor:19]: Still not sure about this, would be great to get a third opinion.

File: Karma.sln [hunk 4]
```diff
@@ -3,7 +3,7 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 16
 VisualStudioVersion = 16.0.31402.337
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Karma", "Karma.csproj", "{FCFA426F-8BB0-445E-A3B8-3489370241A3}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Karma", "Karma.csproj", "{FCFA426F-8BB0-445E-A3B8-3489370241A3}"
```
Comment (ArentasMeinorius) [inline Karma.sln:6]: what is this change?

File: Pages/EditCharityEvent.razor [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+@code {
+    [Parameter]
+    public Guid Id { get; set; }
+    public CharityEvent charityEvent;
+
+    private async Task UpdateEvent()
+    {
+        await dataModificator.UpdateCharityEvent(charityEvent);
+        UriHelper.NavigateTo("");
```
Comment (ArentasMeinorius) [inline Pages/EditCharityEvent.razor:29]: TODO (not for this PR): display alert that item has been modified

File: Pages/ShowIndividualEvent.razor (hunk 3 above)
Comment (ArentasMeinorius) [reply Pages/ShowIndividualEvent.razor:19]: what's wrong with this?

File: Pages/ShowIndividualEvent.razor [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+@code {
+    [Parameter]
+    public Guid Id { get; set; }
+    private CharityEvent charityEvent;
+
+    protected override async Task OnInitializedAsync()
+    {
+        charityEvent = (CharityEvent)(await m_fileContentProvider.ReadFromFileAsync()).Where(p => p.Id == Id).FirstOrDefault();
```
Comment (ArentasMeinorius) [inline Pages/ShowIndividualEvent.razor:28]: default is null, yes?

File: Pages/ShowIndividualEvent.razor [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public void MoveToIndividualEvent()
```
Comment (ArentasMeinorius) [inline Pages/ShowIndividualEvent.razor:31]: would name something different. Maybe NavigateToIndv.. or Display/ShowIndv...

MoveTo (for me) implies that the item is in position X and I will move it to position Y while in reality I will see the item. But feel free to leave the name if you think it fits

File: Pages/ShowIndividualEvent.razor (hunk 3 above)
Comment (Mart1nux) [reply Pages/ShowIndividualEvent.razor:19]: Just had some doubts if it is correct to put an if at a @page section.

Comment (ArentasMeinorius) [reply Pages/ShowIndividualEvent.razor:19]: Oh, never worked with blazor before so I don't know best practices. I'd say let's keep it as it is for now and ask during lecture.

File: Pages/ShowIndividualEvent.razor (hunk 7 above)
Comment (cezaris13) [reply Pages/ShowIndividualEvent.razor:31]: Renamed to `NavigateTo...`

File: Pages/EditCharityEvent.razor (hunk 5 above)
Comment (cezaris13) [reply Pages/EditCharityEvent.razor:29]: Added alert to show that item has been created/updated/deleted

File: Pages/ShowIndividualEvent.razor (hunk 6 above)
Comment (cezaris13) [reply Pages/ShowIndividualEvent.razor:28]: yep

File: Karma.sln (hunk 4 above)
Comment (cezaris13) [reply Karma.sln:6]: It changed automatically after I created a file(I don't remember which one)

**Review by ArentasMeinorius — APPROVED** [review]

File: Pages/ShowIndividualEvent.razor (hunk 6 above)
Comment (ArentasMeinorius) [reply Pages/ShowIndividualEvent.razor:28]: unnecessary cast? if this is for the checklist then leave it

File: Pages/ShowIndividualEvent.razor (hunk 7 above)
Comment (ArentasMeinorius) [reply Pages/ShowIndividualEvent.razor]: i think it would be clearer if you renamed the method as well. But feel free to keep it if you think otherwise

**Pull request merged**